runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -318,7 +331,7 @@ class ExecutionContext final {
 
         // Add a new device
         device->setDeviceStatus(RTDeviceStatus::Active);
-        if (this->seed) {
+        if (this->seed != nullptr) {
```

Comment:
Was there any warning for this? :eyes:

Reply:
Yes, clang-tidy really wants all comparisons to be as explicit as possible

Reply:
Not lately. Last ones reported were:
```
/home/runner/work/pennylane-lightning/pennylane-lightning/Build/pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/../../include/ExecutionContext.hpp:41:7: error: class 'MemoryManager' defines a non-default destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator [cppcoreguidelines-special-member-functions,hicpp-special-member-functions,-warnings-as-errors]
class MemoryManager final {
      ^
/home/runner/work/pennylane-lightning/pennylane-lightning/Build/pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/../../include/ExecutionContext.hpp:54:13: error: do not manage memory manually; use RAII [hicpp-no-malloc,-warnings-as-errors]
            free(allocation);
```

Reply:
@rauletorresc Converted to smart pointers https://github.com/PennyLaneAI/catalyst/pull/1273/commits/de8e1ba30b5752ff4123f012ba306fa22b42db2a